libbeat/processors/dissect/field.go

```diff
@@ -273,6 +354,21 @@ func newIndirectField(id int, key string, length int) indirectField {
 }
 
 func newNormalField(id int, key string, ordinal int, length int, greedy bool) normalField {
+	parts := strings.Split(key, "|")
```

Comment:
Should the extraction of the type be responsibility of `extractKeyParts`?

And maybe `newField` should take care of checking that the type is a valid one.

Reply:
Refactored as suggested above.

Reply:
Extraction of type moved to extractKeyParts, new field checks for valid data type and returns error for invalid, missing data types.